src/AzureAuth/EnvVars.cs

```diff
@@ -23,6 +23,16 @@ internal static class EnvVars
         /// </summary>
         public static readonly string Cache = $"{EnvVarPrefix}_CACHE";
 
+        /// <summary>
+        /// Name of the env var used to disable user based authentication modes.
+        /// </summary>
+        public static readonly string DisableUserBasedAuthentication = $"{EnvVarPrefix}_DISABLE_USER_AUTH";

```

Comment:
Technically "No Interaction" still allows for IWA or Broker in some scenarios. But what we really want is "there is no user - so don't try authing the user at all"

Reply:
That was the original idea to just disable interactivity. Since the non-interactive machines do not need any kind of authentication, specifically user based authentication, we disable user based auth modes which is all modes right now. When we introduce confidential client flows, the non-interactive machines can then use those modes.

Reply:
Made changes to use env variable `DISABLE_INTERACTIVITY`.